### PR TITLE
attribute index structures

### PR DESCRIPTION
**Pull request by kkraune**
first cut, prepare for later _dictionary:hash_ feature. Some questions in ToDo

att

### CODE REVIEW

**Review by geirst — APPROVED** [review]
I think the text in the new box of the illustration should be changed. E.g "Dictionary" or "Dictionary to Posting Lists"

File: en/attributes.html [hunk 1]
```diff
@@ -168,6 +154,47 @@ <h2 id="data-structures">Data structures</h2>
...
+<h2 id="index-structures">Index structures</h2>
+<p>
+  Without <code>fast-search</code>, attribute access is a memory lookup,
+  being one value or all values, depending on query execution.
+  An attribute is a linear array-like data structure -
+  matching documents potentially means scanning <em>all</em> attribute values.
+</p>
+<p>
+  Setting <a href="reference/schema-reference.html#attribute">fast-search</a>
+  creates a an index structure for quicker lookup.
```
Comment (geirst) [inline en/attributes.html:166]: "a an" -> "an"

File: en/attributes.html [hunk 2]
```diff
@@ -168,6 +154,47 @@ <h2 id="data-structures">Data structures</h2>
...
+  This uses more memory, and also more CPU when updating documents.
+  It increases steady state memory usage for all attribute types
+  and also add initialization overhead for numeric types.
+  The default index structure is a b-tree of attribute values pointing to the posting list for the value:
```
[the remark this answers is not in the document]
Comment (kkraune) [reply en/attributes.html:170]: thanks! replaced with another b-tree in the illustration

**Comment by kkraune** [comment]
Thanks @geirst ! We will fine tune this again once we document the dictionary: hash feature shortly

**Pull request merged**